### PR TITLE
feat: add cloudflare r2 support

### PR DESCRIPTION
**Pull request by creaplus**
for solve this issure
https://github.com/casdoor/casdoor/issues/2477

### CODE REVIEW

File: r2/r2_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"fmt"
+	"github.com/casdoor/oss/r2"
+	"github.com/casdoor/oss/tests"
+	"github.com/jinzhu/configor"
+	"testing"
+)
+
+type Config struct {
+	AccountId       string `env:"CREAPLUS_CF_R2_Account_ID"`
```
Comment (hsluoyz) [inline r2/r2_test.go:12]: Don't add your username here

File: r2/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Cloudflare R2
+
+[Cloudflare R2](https://developers.cloudflare.com/r2/) 
+
+## Usage
+
+```go
+import "https://github.com/creaplus/oss"
```
Comment (hsluoyz) [inline r2/README.md:8]: Don't add your username here

File: go.mod [hunk 3]
```diff
@@ -3,20 +3,41 @@ module github.com/casdoor/oss
 go 1.18
 
 require (
+	cloud.google.com/go/storage v1.31.0
 	github.com/Azure/azure-storage-blob-go v0.15.0
 	github.com/aliyun/aliyun-oss-go-sdk v2.2.2+incompatible
 	github.com/aws/aws-sdk-go v1.44.4
+	github.com/aws/aws-sdk-go-v2 v1.22.1
```
Comment (hsluoyz) [inline go.mod:10]: Why a lot of AWS SDKs are added here?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: go.mod (hunk 3 above)
Comment (creaplus) [reply go.mod:10]: Cloudflare provides a demonstration of the use of the SDK. I'll delete it and try to see if it works.

Comment (hsluoyz) [reply go.mod:10]: Cloudflare and AWS are two different companies. We don't need to add AWS SDK to use Cloudflare services.

If the tests require the dependencies, we can remove the tests

Comment (creaplus) [reply go.mod:10]: cloudflare R2 implements S3 API
https://developers.cloudflare.com/r2/api/s3/api/

The current calling method provided by r2 is to use the aws sdk
https://developers.cloudflare.com/r2/examples/aws/aws-sdk-go/

In addition, aws-sdk-go-v2 is a newer sdk version. Do I need to downgrade to v1 version?
https://aws.github.io/aws-sdk-go-v2/docs/migrating/

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@creaplus thanks for the PR. Can you make a PR to casdoor to use the new version?

**Pull request merged**